libtiledbsoma/src/reindexer/reindexer.h

```diff
@@ -53,9 +54,14 @@ class IntIndexer {
      * @param size yhr number of keys in the put
      * @param threads number of threads in the thread pool
      */
-    void map_locations(const int64_t* keys, size_t size, size_t threads = 4);
-    void map_locations(const std::vector<int64_t>& keys, size_t threads = 4) {
-        map_locations(keys.data(), keys.size(), threads);
+    void map_locations(
+        const int64_t* keys,
+        size_t size,
+        std::shared_ptr<tiledbsoma::SOMAContext> context);
```

Comment:
see other comment related to this -- the context should be provided and set during construction, not in a call to map_locations

Reply:
@bkmartinjr I think the code for context looks a bit confusion. Based on a discussion I had with @nguyenv the context is created beforehand and when the call to `pybind` happens we get a copy of the context (as a shared_ptr). So the call to amp_location gets the same context that is passed to `tiledbsoma_build_index(keys, context.native_context)` which is created previously. The only reason to that have to pass the context to map_location is to keep a pointer to it (as lookup doesn't get called with context) and use its threadpool during lookup. @nguyenv does it make sense?

Reply:
To clarify, I mean it should be passed to the IntIndexer during IntIndexer construction.

In other words, with a pre-existing context, code in index_util would look like:
```
    reindexer = IntIndexer(context)
    reindexer.map_locations(keys)
    return reindexer
```

Without this, it allows the possibility that the IntIndexer context is mutable after IntIndexer creation, and I'm pretty sure that is not desirable